staging/src/k8s.io/client-go/rest/url_utils.go

```diff
@@ -33,10 +33,7 @@ func DefaultServerURL(host, apiPath string, groupVersion unversioned.GroupVersio
 	}
 	base := host
 	hostURL, err := url.Parse(base)
-	if err != nil {
-		return nil, "", err
-	}
-	if hostURL.Scheme == "" || hostURL.Host == "" {
+	if err != nil || hostURL.Scheme == "" || hostURL.Host == "" {
```

Comment:
When parsing a url in the form of host:port but without the scheme, url.Parse will set the url.Scheme to `host`, while leave the `url.Host` to empty.